pkg/policies/policies.go

```diff
@@ -177,7 +138,12 @@ func (pv *PolicyVerifier) getMaterialPayload(ctx context.Context, m *v12.Attesta
 	// Use the CAS to look for the material
 	var b bytes.Buffer
 	w := bufio.NewWriter(&b)
-	err := pv.cas.Download(ctx, w, m.GetArtifact().GetDigest())
+
+	client, err := pv.getCASClient(pb.CASCredentialsServiceGetRequest_ROLE_DOWNLOADER, m.GetArtifact().GetDigest())
+	if err != nil {
+		return nil, err
+	}
+	err = client.Download(ctx, w, m.GetArtifact().GetDigest())
```

Comment:
if we happen to do this validation in `attestation add` we would not need to download the file.

Downloading at push when the file is still there doesn't seem optimal and potentially problematic with big files.

Reply:
Yes, I agree.